packages/frontend/src/features/post/components/ActionBtn/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { ReactComponent as BanIcon } from '@/assets/svg/ban.svg'
+import { ReactComponent as CloseIcon } from '@/assets/svg/close-w.svg'
+import { ReactComponent as EllipsisIcon } from '@/assets/svg/ellipsis.svg'
+import { useMediaQuery } from '@uidotdev/usehooks'
+import { ReactEventHandler, useCallback, useState } from 'react'
+
+export default function ActionBtn() {
+    const [isOpen, setIsOpen] = useState(false)
+    const openActionMenu = useCallback(() => setIsOpen(true), [])
+    const closeActionMenu = useCallback(() => setIsOpen(false), [])
+
+    const onClick: ReactEventHandler = (e) => {
+        e.preventDefault()
+        openActionMenu()
+    }
+
+    return (
+        <div className="relative" onClick={(e) => e.preventDefault()}>
```

Comment:
The reason for using preventDefault on the click event handler is that clicking triggers the Link to posts/id that wraps around the Post Component Header. A better approach, in my opinion, would be to modify the structure of the Post Component so that the Link does not wrap the entire header. Once we change the Post component, we can remove preventDefault.